### PR TITLE
Always drop and recreate counters

### PR DESCRIPTION
**Pull request by mikeminutillo**
When generating counter creation scripts, we were checking if the counters already existed. There was a bug in the logic of these scripts. The easiest way to fix this is to not bother checking. If a user runs the generated script, it will always drop and recreate the counters.

### CODE REVIEW

**Comment by dvdstelt** [comment]
Are we 100% sure PerfCounters are retrieved (or worked with) based on their name and not some identifier? I mean, if people add a perfcounter to SCOM and we remove and recreate it, will it still work?

**Comment by VladimirMakaev** [comment]
@mikeminutillo 
If multiple endpoints are hosted on the same box the deployment of any of them will disrupt the work of others. They might not even recover as it could be a critical error. Do you think it's ok?

**Comment by WojcikMike** [comment]
Beyond what @VladimirMakaev wrote.  I don't think it is a good idea. I run the tests of creating Performance Counter Data Collector set. Deleting counters and adding them doesn't impact saved counters at all. However deleting counters causes running endpoints to stop reporting performance metrics. Endpoint needs to be restarted to pick up new counters.

What is more on my machine I had few more instances under `NServiceBus` Performance Counter category: called `audit`, `error`, and 3 other related to service control. I couldn't find any performance counters in them, but I am afraid that some older version of SC might have added Performance Counters into the same category. By deleting category as a whole we might cause some issues.

**Comment by mikeminutillo** [comment]
It's worth noting that the original implementation still dropped and recreated everything if it found a discrepancy.

**Comment by WojcikMike** [comment]
@mikeminutillo true :| So we are keeping the behavior the same. With the difference being if nothing changed script will still recreate the counters. We need to document that running this script require restarting endpoints that use performance counters on the same box.

File: src/ScriptBuilder/CSharpCounterWriter.cs [hunk 1]
```diff
@@ -49,45 +48,18 @@ public static void Create()
...
-                    categoryName: categoryName,
-                    categoryHelp: ""NServiceBus statistics"",
-                    categoryType: PerformanceCounterCategoryType.MultiInstance,
-                    counterData: counterCreationCollection);
-                PerformanceCounter.CloseSharedResources();
-            }}
+            PerformanceCounterCategory.Create(
+                categoryName: categoryName,
+                categoryHelp: ""NServiceBus statistics"",
+                categoryType: PerformanceCounterCategoryType.MultiInstance,
```
Comment (WojcikMike) [inline src/ScriptBuilder/CSharpCounterWriter.cs]: @mikeminutillo @Particular/metrics-maintainers  do you know why we use MultiInstance category type here? Is there any gain in doing that? It seems that we don't use instances in any way.

**Comment by WojcikMike** [comment]
@mikeminutillo I approved the PR as it is equivalent with the previous script. But I believe we should raise a bug and consider making this script less intrusive to cater for scenario of few endpoints running on the same box and having different performance counters

File: src/ScriptBuilder/CSharpCounterWriter.cs (hunk 1 above)
Comment (mikeminutillo) [inline src/ScriptBuilder/CSharpCounterWriter.cs]: We create a single perf counter for Processing Time and then created an instance for each running endpoint. The same for each metric. (I believe this is how it is meant to work but I haven't confirmed it)

**Comment by mikeminutillo** [comment]
@Particular/metrics-maintainers what do I need to do to finish this off?

**Comment by kbaley** [comment]
@mikeminutillo I was going to hold off until #12 was done. But it seems to be lingering. Will merge.

**Comment by mikeminutillo** [comment]
@kbaley cool. When you know which release this will go into, can you update #9 accordingly?

**Pull request merged**